test/unit/workspace/datastore_test.py

```diff
@@ -335,4 +338,99 @@ def test_get_resource_with_nonexistent_resource_fails(self):
         self.assertRaises(KeyError, self.fetcher.get_resource, res)
 
 
+def test_get_zipfile_resource_retry(mocker):
+    ds = datastore.Datastore()
+    not_a_zipfile = b"aaa"
+    ds.get_unique_resource = mocker.MagicMock(return_value=not_a_zipfile)
```

Comment:
Potentially better to patch the `zipfile.ZipFile` method so we can easily test the "error, then succeed" flow.